Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM ruby:2.4
+
+RUN apt-get update \
+  && apt-get install -y \
+    git \
+    make
+
+COPY . /gh-pages
+WORKDIR /gh-pages
+
+RUN bundle config local.github-pages /gh-pages
+RUN bundle install
```

Comment:
We may need to modify the Gemfile to add the Gem to the `jekyll_plugins` group so that it can hook earlier into the bootstrap process and set defaults, e.g., `gem "github-pages", group: :jekyll_plugins`

Reply:
I'm not sure I totally understand what you mean with this, please elaborate and I'll try to get on it asap :wink:.

My GitHub Pages projects do contain the `gem "github-pages", group: :jekyll_plugins` line, but then I believe I should then probably fire up "bundle exec jekyll server ..." instead of just  "jekyll serve ...".

Is there any logic performed on the `jekyll_plugins` group? Is it not only there for organization?

Reply:
> Is there any logic performed on the jekyll_plugins group? Is it not only there for organization?

The Jekyll Plugins group allows the GitHub Pages Gem to get bootstrapped earlier in Jekyll's process and set configuration defaults and constraints to match GitHub Pages. I'm struggling to load a gemspec into a gem group, so we may need to create a new / modify the Gemfile for docker.

Reply:
Have you guys figures this out in the end? It seems to me that jekyll in the container ignores github-pages entirely and just uses its own default configuration, making this whole `make server` and `contrib/func.sh` thing rather useless. :-/